# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCMarker.h
# Commit: 4746d15d7a9f
# Full Hash: 4746d15d7a9fa501c702bd1437d3f3da127a86ee
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-28 04:43:55
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 3: Remove the unused MarkStackIter class r=sfink
#   
#   Depends on D160525
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160526
# ==============================================================================

diff -r 8be86d9e2f18 -r 4746d15d7a9f js/src/gc/GCMarker.h
--- a/js/src/gc/GCMarker.h	Thu Oct 27 18:23:02 2022 +0000
+++ b/js/src/gc/GCMarker.h	Thu Oct 27 18:23:03 2022 +0000
@@ -204,28 +204,6 @@
 #ifdef DEBUG
   mutable size_t iteratorCount_;
 #endif
-
-  friend class MarkStackIter;
-};
-
-class MarkStackIter {
-  MarkStack& stack_;
-  size_t pos_;
-
- public:
-  explicit MarkStackIter(MarkStack& stack);
-  ~MarkStackIter();
-
-  bool done() const;
-  MarkStack::Tag peekTag() const;
-  MarkStack::TaggedPtr peekPtr() const;
-  MarkStack::SlotsOrElementsRange peekSlotsOrElementsRange() const;
-  void next();
-  void nextPtr();
-  void nextArray();
-
- private:
-  size_t position() const;
 };
 
 } /* namespace gc */